apps/music/test/marionette/playlist_test.js

```diff
@@ -154,16 +155,89 @@ marionette('Music player playlist', function() {
-      assert.equal(songs[0].findElement('.list-song-title').text(),
-                   'Yield to thread');
+      assert.equal(PlaylistHelper.songIndex(songs[0]), '1.01');
+      assert.equal(PlaylistHelper.songTitle(songs[0]), 'Yield to thread');
 
-      assert.equal(songs[2].findElement('.list-song-index').text(), '1.03');
-      assert.equal(songs[2].findElement('.list-song-title').text(),
-                   'Windows BSOD');
+      assert.equal(PlaylistHelper.songIndex(songs[2]), '1.03');
+      assert.equal(PlaylistHelper.songTitle(songs[2]), 'Windows BSOD');
 
-      assert.equal(songs[3].findElement('.list-song-index').text(), '2.01');
-      assert.equal(songs[3].findElement('.list-song-title').text(), 'Crash');
+      assert.equal(PlaylistHelper.songIndex(songs[3]), '2.01');
+      assert.equal(PlaylistHelper.songTitle(songs[3]), 'Crash');
     });
   });
+
+  suite('Highest rated playlist', function() {
+    setup(function() {
+      client.fileManager.removeAllFiles();
+      client.fileManager.add([
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/a.ogg'
+        },
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/b.ogg'
+        },
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/c.ogg'
+        },
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/w.ogg'
+        },
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/x.ogg'
+        },
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/y.ogg'
+        },
+      ]);
+    });
+
+    test('Rating sort order', function() {
+      music.launch();
+      music.waitForFirstTile();
+
+      music.switchToAlbumsView();
+
+      music.selectAlbum('We crash computers');
+
+      music.waitForSongs(function(songs) {
+        return songs.length >= 6;
+      });
+
+      var songs = music.songs;
+      assert.equal(songs.length, 6);
+
+      client.executeScript(function() {
+        var w = window.wrappedJSObject;
+        var songData = w.SubListView.dataSource[3];
+        songData.metadata.rated = 4;
+        w.musicdb.updateMetadata(songData.name, songData.metadata);
+
+        songData = w.SubListView.dataSource[1];
+        songData.metadata.rated = 5;
+        w.musicdb.updateMetadata(songData.name, songData.metadata);
+      });
+
+      music.switchToPlaylistsView();
+
+      music.selectPlaylist('Highest rated');
+
+      music.waitForSongs(function(songs) {
+        return songs.length >= 6;
+      });
+      songs = music.songs;
+
+      assert.equal(PlaylistHelper.songIndex(songs[0]), '1');
+      assert.equal(PlaylistHelper.songTitle(songs[0]), 'XOXO');
+
+      assert.equal(PlaylistHelper.songIndex(songs[1]), '2');
+      assert.equal(PlaylistHelper.songTitle(songs[1]), 'Crash');
+    });
```

Comment:
This looks good to me. My only comment is that it is hard to figure out what is supposed to happen because the files are listed in a different order than the songs appear in the album. So when you give ratings for songs 1 and 3, I can't tell which songs those are without running 'strings' over the .ogg binaries.

I think you could clarify this by modifying your fileManager.add() call to list the files in the order the songs appear on the album, and also to add coments like "// Track 1.02: XOXO" 

That is just a nit, however and you can land without it if you don't think it is worth it.